user/libpinyin/template.py

```diff
@@ -0,0 +1,35 @@
+pkgname = "libpinyin"
+pkgver = "2.9.92"
+pkgrel = 0
+build_style = "gnu_configure"
+configure_args = [
+    "--with-dbm=KyotoCabinet", 
+    "--enable-libzhuyin"
+]
+hostmakedepends = [
+    "pkgconf", 
+    "autoconf", 
+    "automake", 
+    "libtool",
+    "wget2",
+    "gettext"
+]
+
+makedepends = [
+    "kyotocabinet-devel", 
+    "glib-devel"
+]
+    
```

Comment:
```suggestion
```

Reply:
(that means to delete the line; we have no newlines between these variables normally, it's just one contiguous block)

Reply:
okay sir

Reply:
the newlines are still there around makedepends

Reply:
> i only removed whitespaces by hand initially, and they passed ruff check

`ruff check` and `ruff format` are different and separate things; they check different stuff.

the ci runs both

---

also, these newlines are still there, and you haven't removed them still. do you perhaps not understand what i mean?

currently the code is like this:

```
hostmakedepends = ["pkgconf", "automake", "libtool", "gettext"]

makedepends = ["kyotocabinet-devel", "glib-devel"]

pkgdesc = "Algorithms core for intelligent sentence-based Chinese pinyin IME"
```

it should be like this:

```
hostmakedepends = ["pkgconf", "automake", "libtool", "gettext"]
makedepends = ["kyotocabinet-devel", "glib-devel"]
pkgdesc = "Algorithms core for intelligent sentence-based Chinese pinyin IME"
```

i.e. the extra empty lines should not be there. ruff format does not check this specifically, but the newlines should not be there

Reply:
Just removed them. sorry for making so many unnecessary commits. Will they get all pushed to the main repo, or get merged into one commit eventually?

Reply:
you accidentally removed the trailing newline at the bottom of the file while changing that, somehow. that one is not meant to be removed

> Will they get all pushed to the main repo, or get merged into one commit eventually?

you have to squash them to two final commits yourself- one commit per package added, i.e. `user/kyotocabinet: new package` and `user/libpinyin: new package`

Reply:
> you accidentally removed the trailing newline at the bottom of the file while changing that, somehow. that one is not meant to be removed
> 
> > Will they get all pushed to the main repo, or get merged into one commit eventually?
> 
> you have to squash them to two final commits yourself- one commit per package added, i.e. `user/kyotocabinet: new package` and `user/libpinyin: new package`

unfortunately, I edited both packages' files in one commit before, so the best I can do is to merge all those commits into one commit, is that okay? contributing.md said it's fine, but i'm not so sure

Reply:
it doesn't say it's fine, you need two commits

doing it is easy, you can `git reset master` and then just make two invidivual commits and push -f

Reply:
finally! they're good to go.